### PR TITLE
Dockerizing

### PR DESCRIPTION
**Pull request by jango2106**
## Proposed Changes
- Dockerized the ui and api projects
- Dockerized an instance of the mongo db so that we can have a local instance running for testing.

## Testing
- Pull down project
- Follow ReadMe instructions for starting up the project (add a -d to the up command to run it in a detatched mode)
- Make sure all of the docker containers start in an ok status by running `docker ps`. There should be 3 containers.
- Do a GET request to `localhost:8080/api/v1/strands` should return `[]` with 200 status.
- POST request to `localhost:8080/api/v1/strands` with body below should return a 201 created status.
```json
{
	"ipAddress":"127.0.0.1",
	"name": "TestStrand",
	"port": 1234
}
```
- Do the same GET request as above and you should see one strand.
- Do a `docker-compose down` to wipe away all of the containers.
- Do another `docker-compose up` to create new containers.
- Do the same GET request to see that the data persisted.

### CODE REVIEW

File: server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM openjdk:8 as builder
+
+COPY . /server
+WORKDIR /server
+
+RUN chmod 777 mvnw
+
+# Running from Windows will add a windows EOF. Need to change it to linux.
+RUN /bin/sed -i 's/\r//' mvnw
```
Comment (ahoopes16) [inline server/Dockerfile:9]: Silly Windows

File: server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+# Build Spring project
+RUN ./mvnw install
+
+FROM openjdk:8
```
Comment (ahoopes16) [inline server/Dockerfile:14]: How come you need this a second time?

File: server/Dockerfile (hunk 1 above)
Comment (jango2106) [reply server/Dockerfile:9]: Right!

File: server/Dockerfile (hunk 2 above)
Comment (jango2106) [reply server/Dockerfile:14]: The first one is an intermediate container. That way we can pull the whole project in do all build things in the first one. And then the 2nd one is just the java environment and the jar

**Pull request merged**